src/apps/eclipse/mosquitto/mqtt/mode/numericvalue.pm

```diff
@@ -0,0 +1,205 @@
+        $value = eval "$value $self->{instance_mode}{option_results}->{format_custom}";
+    }
+
+    return sprintf($format, $value);
+}
+
+sub custom_generic_perfdata {
+    my ($self, %options) = @_;
+
+    $self->{output}->perfdata_add(
+        label    => $options{option_results}->{perfdata_name},
+        unit     => $options{option_results}->{perfdata_unit},
+        value    => $self->{result_values}->{numericvalue},
+        warning  => $self->{perfdata}->get_perfdata_for_output(label => 'warning-' . $self->{thlabel}),
+        critical => $self->{perfdata}->get_perfdata_for_output(label => 'critical-' . $self->{thlabel}),
+        min      => $options{option_results}->{perfdata_min},
+        max      => $options{option_results}->{perfdata_max}
+    );
+}
+
+sub custom_generic_threshold {
+    my ($self, %options) = @_;
+
+    return $self->{perfdata}->threshold_check(
+        value     => $self->{result_values}->{numericvalue},
+        threshold => [
+            { label => 'critical-' . $self->{thlabel}, exit_litteral => 'critical' },
+            { label => 'warning-' . $self->{thlabel}, exit_litteral => 'warning' },
+            { label => 'unknown-' . $self->{thlabel}, exit_litteral => 'unknown' }
+        ]
+    );
+}
+
+sub set_counters {
+    my ($self, %options) = @_;
+
+    $self->{maps_counters_type} = [
+        { name => 'global', type => 0 }
+    ];
+
+    $self->{maps_counters}->{global} = [
+        { label => 'generic',
+          set   => {
+              key_values                     => [{ name => 'numericvalue' }],
+              closure_custom_output          => $self->can('custom_generic_output'),
+              closure_custom_perfdata        => $self->can('custom_generic_perfdata'),
+              closure_custom_threshold_check => $self->can('custom_generic_threshold')
+          }
+        }
+    ];
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::check_options(%options);
+
+    if (centreon::plugins::misc::is_empty($options{option_results}->{topic})) {
+        $self->{output}->add_option_msg(short_msg => 'Missing parameter --topic.');
+        $self->{output}->option_exit();
+    }
+}
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    my $value = $options{mqtt}->query(
+        topic => $self->{option_results}->{topic}
+    );
+
+    if (!centreon::plugins::misc::is_empty($self->{option_results}->{extracted_pattern})) {
+        if ($value =~ /$self->{option_results}->{extracted_pattern}/ && defined($1)) {
+            $value = $1;
+        }
+    }
+    if ($value !~ /^-?\d+(?:\.\d+)?$/) {
+        $self->{output}->output_add(
+            severity  => 'UNKNOWN',
+            short_msg => 'topic value is not numeric (' . $value . ')'
+        );
+        return;
+    }
+
+    if (!defined($value)) {
+        $self->{output}->add_option_msg(short_msg => "Cannot find information");
```

Comment:
For troubleshooting purpose, I would have preferred that the message reads "Could not find a numeric value in '$value'" but the evaluated string from where we try to get it is lost at this point